### ISSUE TITLE
Incorrectly identified circular reference

### ISSUE THREAD

**Issue by sgoodgrove** [EPPlusSoftware/EPPlus#905]
This is a little tricky to explain in English, but when an expression refers to a cell on another sheet with the same row/column as a cell that is used on the originating expression a circular reference is incorrectly flagged and a CircularReferenceException is throw.

I've produced a unit test below that succinctly demonstrates the issue. I believe this is happening because when the expression chain is being followed the dependency sheet context is not updated, but is instead retained form the first sheet in the dependency chain.

N.B. This ticket is possibly a duplicate of #795

# Test Case Code

```csharp
using NUnit.Framework;

using OfficeOpenXml;

namespace UnitTests.Spreadsheet
{
    [TestFixture]
    public class CircularReferenceTest
    {
        [Test]
        public void Test1()
        {
            var package = new ExcelPackage();
            var sheet1 = package.Workbook.Worksheets.Add("Sheet1");
            var sheet2 = package.Workbook.Worksheets.Add("Sheet2");

            sheet1.Cells["A1"].Value = 1d;
            sheet1.Cells["A2"].Value = 3d;
            sheet1.Cells["A3"].Formula = "=Sheet2!A1";

            sheet2.Cells["A1"].Formula = "=Sheet1!A1";

            CalculationExtension.Calculate(package.Workbook);
        }
    }
}
```

If I were to speculate, I would suggest that maybe the cause is in `DependencyChainFactory.cs` in the `FollowChain()` method and when `f` is pushed onto the stack and popped off of the stack that `ws` should perhaps be set to `f.ws` and then when checking for the collision instead of doing `adr.Collide(f.Row, f.Column, f.Row, f.Column)` do `adr.Collide(new ExcelBaseAddress(f.ws.Name, f.Row, f.Column, f.Row, f.Column)` or similar.

**Comment by JanKallman** [comment]
This looks like a bug in EPPlus 6. It seems to work as expected in EPPlus 7 preview 1.
I'll see if we can provide a fix for it.

**Comment by JanKallman** [comment]
This should be fixed in version 6.2.5
